pycbc/waveform/generator.py

```diff
@@ -335,6 +357,32 @@ def _postgenerate(self, res):
         return hp, hc
 
 
+class TDomainCBCModesGenerator(BaseCBCGenerator):
+    """Generates time domain CBC waveform modes in the radiation frame.
+
+    Uses :py:func:`waveform_modes.get_td_waveform_modes` as a generator
+    function to create time-domain CBC waveforms in the radiation frame; i.e.,
+    with no detector response function applied. For more details, see
+    :py:class:`BaseGenerator`.
+    """
+    def __init__(self, variable_args=(), **frozen_params):
+        super(TDomainCBCModesGenerator, self).__init__(
+            waveform_modes.get_td_waveform_modes,
+            variable_args=variable_args, **frozen_params)
+
+    def _postgenerate(self, res):
+        """Applies a taper if it is in current params.
+        """
+        if 'taper' in self.current_params:
+            tapermethod = self.current_params['taper']
+            for mode in res:
+                hp, hc = res[mode]
```

Comment:
I think these are Re[Hlm] and Im[Hlm] and not hp and hc ... right? If so, please change the variable name to make this clear.

Reply:
Good catch... yup, you're right. Renamed to "ulm" and "vlm", which is the naming I used in the waveform modes module.